pkg/ottl/expression.go

```diff
@@ -406,6 +406,24 @@ func (g StandardFunctionGetter[K]) Get(args Arguments) (Expr[K], error) {
 	return Expr[K]{exprFunc: fn}, nil
 }
 
+type PMapGetSetter[K any] interface {
```

Comment:
Could you please move these OTTL changes to a separate PR? It would also be nice to have some unit tests for that as well. Thanks!

Reply:
@edmocosta The code has been split out into https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39657. I still leave it here and fix things up when  #39657 has been merged into main.